FATAL:  database "repmgr" does not exist

### Name and Version

bitnami/postgresql-ha:14.0.10

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Kubernetes version: v1.28.6
Helm: v3.13.1


### Are you using any custom parameters or values?
```
--set global.postgresql.username=user1 \
--set global.postgresql.database=db1 \
--set postgresql.audit.logConnections=true \
--set postgresql.audit.logDisconnections=true \
--set pgpool.logConnections=true \
--set pgpool.logPerNodeStatement=true \
--set witness.create=true \
--set witness.audit.logConnections=true \
--set witness.audit.logDisconnections=true \
--set metrics.enabled=true \
--set metrics.serviceMonitor.enabled=true \
--set persistence.enabled=false
```
### What do you see instead?
```
postgresql-repmgr 11:03:31.32 INFO  ==> 
postgresql-repmgr 11:03:31.33 INFO  ==> Welcome to the Bitnami postgresql-repmgr container
postgresql-repmgr 11:03:31.33 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
postgresql-repmgr 11:03:31.33 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
postgresql-repmgr 11:03:31.34 INFO  ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit https://bitnami.com/enterprise
postgresql-repmgr 11:03:31.34 INFO  ==> 
postgresql-repmgr 11:03:31.43 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
postgresql-repmgr 11:03:31.54 INFO  ==> Validating settings in REPMGR_* env vars...
postgresql-repmgr 11:03:31.55 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql-repmgr 11:03:31.56 INFO  ==> Querying all partner nodes for common upstream node...
postgresql-repmgr 11:03:31.82 INFO  ==> There are no nodes with primary role. Assuming the primary role...
postgresql-repmgr 11:03:31.84 INFO  ==> Preparing PostgreSQL configuration...
postgresql-repmgr 11:03:31.84 INFO  ==> postgresql.conf file not detected. Generating it...
postgresql-repmgr 11:03:32.25 INFO  ==> Preparing repmgr configuration...
postgresql-repmgr 11:03:32.34 INFO  ==> Initializing Repmgr...
postgresql-repmgr 11:03:32.34 INFO  ==> Initializing PostgreSQL database...
postgresql-repmgr 11:03:32.35 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql-repmgr 11:03:32.42 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
postgresql-repmgr 11:03:32.52 INFO  ==> Deploying PostgreSQL with persisted data...
postgresql-repmgr 11:03:32.63 INFO  ==> Configuring replication parameters
postgresql-repmgr 11:03:32.75 INFO  ==> Configuring fsync

postgresql-repmgr 11:03:32.83 INFO  ==> ** PostgreSQL with Replication Manager setup finished! **
postgresql-repmgr 11:03:32.94 INFO  ==> Starting PostgreSQL in background...
waiting for server to start....2024-05-10 11:03:33.062 GMT [153] LOG:  pgaudit extension initialized
2024-05-10 11:03:33.128 GMT [153] LOG:  redirecting log output to logging collector process
2024-05-10 11:03:33.128 GMT [153] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
2024-05-10 11:03:33.128 GMT [153] LOG:  starting PostgreSQL 16.2 on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
2024-05-10 11:03:33.129 GMT [153] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2024-05-10 11:03:33.129 GMT [153] LOG:  listening on IPv6 address "::", port 5432
2024-05-10 11:03:33.141 GMT [153] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2024-05-10 11:03:33.160 GMT [157] LOG:  database system was shut down at 2024-05-10 11:03:30 GMT
2024-05-10 11:03:33.172 GMT [153] LOG:  database system is ready to accept connections
 done
server started
2024-05-10 11:03:33.243 GMT [163] LOG:  connection received: host=[local]
postgresql-repmgr 11:03:33.24 INFO  ==> ** Starting repmgrd **
[2024-05-10 11:03:33] [NOTICE] repmgrd (repmgrd 5.4.1) starting up
2024-05-10 11:03:33.264 GMT [166] LOG:  connection received: host=postgresql-ha-postgresql-0.postgresql-ha-postgresql-headless.default.svc.cluster.local port=32870
2024-05-10 11:03:33.340 GMT [166] LOG:  connection authenticated: identity="repmgr" method=md5 (/opt/bitnami/postgresql/conf/pg_hba.conf:1)
2024-05-10 11:03:33.340 GMT [166] LOG:  connection authorized: user=repmgr database=repmgr application_name=repmgr
2024-05-10 11:03:33.340 GMT [166] FATAL:  database "repmgr" does not exist
[2024-05-10 11:03:33] [ERROR] connection to database failed
[2024-05-10 11:03:33] [DETAIL] 
connection to server at "postgresql-ha-postgresql-0.postgresql-ha-postgresql-headless.default.svc.cluster.local" (10.233.96.193), port 5432 failed: FATAL:  database "repmgr" does not exist

[2024-05-10 11:03:33] [DETAIL] attempted to connect using:
  user=repmgr password=Iqo2VU54uk connect_timeout=5 dbname=repmgr host=postgresql-ha-postgresql-0.postgresql-ha-postgresql-headless.default.svc.cluster.local port=5432 fallback_application_name=repmgr options=-csearch_path=
```
### Additional information

If I remove these parameters everything starts working
- global.postgresql.username=user1
- global.postgresql.database=db1

Ok.  Sorry for the imformation overload, but I assume more data is better than less.

I systematically deployed chart 15.3.15 with the values.yaml settings in the above comment and between each helm delete I deleted the pvc to ensure clean deploys and only updated the tag to change the container version.  Then I compared and noted the failures.  Here is the list where 'SAME' means the fail was the same as the last one.

- 17.2.0-debian-12-r0 - works
- 17.2.0-debian-12-r1 - works
- 17.2.0-debian-12-r2 - FATAL:  database "repmgr" does not exist
- 17.2.0-debian-12-r3 - SAME
- 17.2.0-debian-12-r4 - SAME
- 17.2.0-debian-12-r5 - SAME
- 17.2.0-debian-12-r6 - SAME
- 17.2.0-debian-12-r7 - SAME
- 17.2.0-debian-12-r8 - SAME
- 17.2.0-debian-12-r9 - FATAL:  password authentication failed for user "repmgr"
								< 2025-05-06 15:51:19.832 UTC [unknown] repmgr repmgr 178 681a2ff7.b2 2025-05-06 15:51:19 UTC 10.42.0.92(37414) >DETAIL:  Role "repmgr" does not exist.
										Connection matched file "/opt/bitnami/postgresql/conf/pg_hba.conf" line 9: "hostnossl    all             all             all                     md5"
								[2025-05-06 15:51:19] [ERROR] connection to database failed
								[2025-05-06 15:51:19] [DETAIL] connection to server at "infrastructure-postgresql-ha-postgresql-0.infrastructure-postgresql-ha-postgresql-headless.default.svc.cluster.local" (10.42.0.92), port 5432 failed: FATAL:  password authentication failed for user "repmgr"
									password retrieved from file "/opt/bitnami/repmgr/conf/.pgpass"
								[2025-05-06 15:51:19] [DETAIL] attempted to connect using:
									user=repmgr passfile=/opt/bitnami/repmgr/conf/.pgpass connect_timeout=5 dbname=repmgr host=infrastructure-postgresql-ha-postgresql-0.infrastructure-postgresql-ha-postgresql-headless.default.svc.cluster.local port=5432 fallback_application_name=repmgr options=-csearch_path=

- 17.3.0-debian-12-r0 - SAME
- 17.3.0-debian-12-r1 - SAME
- 17.3.0-debian-12-r2 - SAME
- 17.3.0-debian-12-r3 - SAME
- 17.3.0-debian-12-r4 - SAME
- 17.4.0-debian-12-r0 - SAME
- 17.4.0-debian-12-r1 - SAME
- 17.4.0-debian-12-r2 - SAME
- 17.4.0-debian-12-r3 - SAME
- 17.4.0-debian-12-r4 - SAME
- 17.4.0-debian-12-r5 - SAME
- 17.4.0-debian-12-r6 - SAME
- 17.4.0-debian-12-r7 - SAME
- 17.4.0-debian-12-r8 - SAME
- 17.4.0-debian-12-r9 - SAME
- 17.4.0-debian-12-r10 - SAME
- 17.4.0-debian-12-r11 - SAME
- 17.4.0-debian-12-r12 - SAME
- 17.4.0-debian-12-r13 - SAME
- 17.4.0-debian-12-r14 - SAME
- 17.4.0-debian-12-r15 - SAME
- 17.4.0-debian-12-r16 - SAME


Here is the order of logs.  The primary pod will come up with a set of logs then RESTART with a second set of logs.


*******************FIRST ********************
postgresql-repmgr 15:59:22.81 INFO  ==>
postgresql-repmgr 15:59:22.82 INFO  ==> Welcome to the Bitnami postgresql-repmgr container
postgresql-repmgr 15:59:22.82 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
postgresql-repmgr 15:59:22.82 INFO  ==> Did you know there are enterprise versions of the Bitnami catalog? For enhanced secure software supply chain features, unlimited pulls from Docker, LTS support, or application customization, see Bitnami Premium or Tanzu Application Catalog. See https://www.arrow.com/globalecs/na/vendors/bitnami/ for more information.
postgresql-repmgr 15:59:22.82 INFO  ==>
postgresql-repmgr 15:59:22.84 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
postgresql-repmgr 15:59:22.87 INFO  ==> Validating settings in REPMGR_* env vars...
postgresql-repmgr 15:59:22.88 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql-repmgr 15:59:22.89 INFO  ==> Querying all partner nodes for common upstream node...
postgresql-repmgr 15:59:23.03 INFO  ==> There are no nodes with primary role. Assuming the primary role...
postgresql-repmgr 15:59:23.04 INFO  ==> Preparing PostgreSQL configuration...
postgresql-repmgr 15:59:23.04 INFO  ==> postgresql.conf file not detected. Generating it...
postgresql-repmgr 15:59:23.16 INFO  ==> Preparing repmgr configuration...
postgresql-repmgr 15:59:23.17 INFO  ==> Initializing Repmgr...
postgresql-repmgr 15:59:23.18 INFO  ==> Initializing PostgreSQL database...
postgresql-repmgr 15:59:23.20 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql-repmgr 15:59:23.21 INFO  ==> pg_hba.conf file not detected. Generating it...
postgresql-repmgr 15:59:23.21 INFO  ==> Generating local authentication configuration
postgresql-repmgr 15:59:24.01 INFO  ==> Starting PostgreSQL in background...
postgresql-repmgr 15:59:25.09 INFO  ==> Creating user beast
postgresql-repmgr 15:59:25.10 INFO  ==> Granting access to "beast" to the database "beast"
postgresql-repmgr 15:59:25.13 INFO  ==> Setting ownership for the 'public' schema database "beast" to "beast"
postgresql-repmgr 15:59:25.15 INFO  ==> Stopping PostgreSQL...
waiting for server to shut down.... done
server stopped
postgresql-repmgr 15:59:25.38 INFO  ==> Configuring replication parameters
postgresql-repmgr 15:59:25.41 INFO  ==> Configuring fsync
postgresql-repmgr 15:59:25.41 INFO  ==> Starting PostgreSQL in background...
postgresql-repmgr 15:59:26.46 INFO  ==> Creating repmgr user: repmgr
postgresql-repmgr 15:59:26.49 INFO  ==> Stopping PostgreSQL...
waiting for server to shut down.... done
server stopped

*********************SECOND ************************
postgresql-repmgr 16:00:25.49 INFO  ==>
postgresql-repmgr 16:00:25.50 INFO  ==> Welcome to the Bitnami postgresql-repmgr container
postgresql-repmgr 16:00:25.50 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
postgresql-repmgr 16:00:25.50 INFO  ==> Did you know there are enterprise versions of the Bitnami catalog? For enhanced secure software supply chain features, unlimited pulls from Docker, LTS support, or application customization, see Bitnami Premium or Tanzu Application Catalog. See https://www.arrow.com/globalecs/na/vendors/bitnami/ for more information.
postgresql-repmgr 16:00:25.50 INFO  ==>
postgresql-repmgr 16:00:25.51 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
postgresql-repmgr 16:00:25.53 INFO  ==> Validating settings in REPMGR_* env vars...
postgresql-repmgr 16:00:25.55 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql-repmgr 16:00:25.57 INFO  ==> Querying all partner nodes for common upstream node...
postgresql-repmgr 16:00:25.75 INFO  ==> There are no nodes with primary role. Assuming the primary role...
postgresql-repmgr 16:00:25.75 INFO  ==> Preparing PostgreSQL configuration...
postgresql-repmgr 16:00:25.75 INFO  ==> postgresql.conf file not detected. Generating it...
postgresql-repmgr 16:00:25.86 INFO  ==> Preparing repmgr configuration...
postgresql-repmgr 16:00:25.87 INFO  ==> Initializing Repmgr...
postgresql-repmgr 16:00:25.88 INFO  ==> Initializing PostgreSQL database...
postgresql-repmgr 16:00:25.88 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql-repmgr 16:00:25.90 INFO  ==> pg_hba.conf file not detected. Generating it...
postgresql-repmgr 16:00:25.90 INFO  ==> Generating local authentication configuration
postgresql-repmgr 16:00:25.92 INFO  ==> Deploying PostgreSQL with persisted data...
postgresql-repmgr 16:00:25.94 INFO  ==> Configuring replication parameters
postgresql-repmgr 16:00:25.96 INFO  ==> Configuring fsync
postgresql-repmgr 16:00:25.97 INFO  ==> Applying custom /opt/bitnami/postgresql/conf/pg_hba.conf
postgresql-repmgr 16:00:25.97 INFO  ==> ** PostgreSQL with Replication Manager setup finished! **

postgresql-repmgr 16:00:25.99 INFO  ==> Starting PostgreSQL in background...
pg_ctl: another server might be running; trying to start server anyway
server starting
< 2025-05-06 16:00:26.029 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >LOG:  pgaudit extension initialized
< 2025-05-06 16:00:26.040 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >LOG:  redirecting log output to logging collector process
< 2025-05-06 16:00:26.040 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
< 2025-05-06 16:00:26.040 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >LOG:  starting PostgreSQL 17.3 on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
< 2025-05-06 16:00:26.041 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >LOG:  listening on IPv4 address "0.0.0.0", port 5432
< 2025-05-06 16:00:26.041 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >LOG:  listening on IPv6 address "::", port 5432
< 2025-05-06 16:00:26.043 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
< 2025-05-06 16:00:26.044 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >LOG:  hostssl record cannot match because SSL is disabled
< 2025-05-06 16:00:26.044 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >HINT:  Set "ssl = on" in postgresql.conf.
< 2025-05-06 16:00:26.044 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >CONTEXT:  line 10 of configuration file "/opt/bitnami/postgresql/conf/pg_hba.conf"
< 2025-05-06 16:00:26.046 UTC    169 681a321a.a9 2025-05-06 16:00:26 UTC  >LOG:  database system was interrupted; last known up at 2025-05-06 15:59:55 UTC
< 2025-05-06 16:00:26.243 UTC    169 681a321a.a9 2025-05-06 16:00:26 UTC  >LOG:  database system was not properly shut down; automatic recovery in progress
< 2025-05-06 16:00:26.245 UTC    169 681a321a.a9 2025-05-06 16:00:26 UTC  >LOG:  invalid record length at 0/3000118: expected at least 24, got 0
< 2025-05-06 16:00:26.245 UTC    169 681a321a.a9 2025-05-06 16:00:26 UTC  >LOG:  redo is not required
< 2025-05-06 16:00:26.247 UTC    167 681a321a.a7 2025-05-06 16:00:26 UTC  >LOG:  checkpoint starting: end-of-recovery immediate wait
< 2025-05-06 16:00:26.251 UTC    167 681a321a.a7 2025-05-06 16:00:26 UTC  >LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sync=0.002 s, total=0.005 s; sync files=2, longest=0.001 s, average=0.001 s; distance=0 kB, estimate=0 kB; lsn=0/3000118, redo lsn=0/3000118
< 2025-05-06 16:00:26.254 UTC    163 681a321a.a3 2025-05-06 16:00:26 UTC  >LOG:  database system is ready to accept connections
< 2025-05-06 16:00:27.029 UTC [unknown] [unknown] [unknown] 175 681a321b.af 2025-05-06 16:00:27 UTC 127.0.0.1(52008) >LOG:  connection received: host=127.0.0.1 port=52008
< 2025-05-06 16:00:27.029 UTC [unknown] postgres postgres 175 681a321b.af 2025-05-06 16:00:27 UTC 127.0.0.1(52008) >LOG:  connection authenticated: user="postgres" method=trust (/opt/bitnami/postgresql/conf/pg_hba.conf:3)
< 2025-05-06 16:00:27.029 UTC [unknown] postgres postgres 175 681a321b.af 2025-05-06 16:00:27 UTC 127.0.0.1(52008) >LOG:  connection authorized: user=postgres database=postgres application_name=pg_isready
< 2025-05-06 16:00:27.043 UTC pg_isready postgres postgres 175 681a321b.af 2025-05-06 16:00:27 UTC 127.0.0.1(52008) >LOG:  disconnection: session time: 0:00:00.014 user=postgres database=postgres host=127.0.0.1 port=52008
postgresql-repmgr 16:00:27.04 INFO  ==> ** Starting repmgrd **
[2025-05-06 16:00:27] [NOTICE] repmgrd (repmgrd 5.5.0) starting up
< 2025-05-06 16:00:27.063 UTC [unknown] [unknown] [unknown] 178 681a321b.b2 2025-05-06 16:00:27 UTC 10.42.0.113(49516) >LOG:  connection received: host=10.42.0.113 port=49516
< 2025-05-06 16:00:27.075 UTC [unknown] repmgr repmgr 178 681a321b.b2 2025-05-06 16:00:27 UTC 10.42.0.113(49516) >FATAL:  password authentication failed for user "repmgr"
< 2025-05-06 16:00:27.075 UTC [unknown] repmgr repmgr 178 681a321b.b2 2025-05-06 16:00:27 UTC 10.42.0.113(49516) >DETAIL:  Role "repmgr" does not exist.
        Connection matched file "/opt/bitnami/postgresql/conf/pg_hba.conf" line 9: "hostnossl    all             all             all                     md5"
[2025-05-06 16:00:27] [ERROR] connection to database failed
[2025-05-06 16:00:27] [DETAIL]
connection to server at "infrastructure-postgresql-ha-postgresql-0.infrastructure-postgresql-ha-postgresql-headless.default.svc.cluster.local" (10.42.0.113), port 5432 failed: FATAL:  password authentication failed for user "repmgr"
password retrieved from file "/opt/bitnami/repmgr/conf/.pgpass"

[2025-05-06 16:00:27] [DETAIL] attempted to connect using:
  user=repmgr passfile=/opt/bitnami/repmgr/conf/.pgpass connect_timeout=5 dbname=repmgr host=infrastructure-postgresql-ha-postgresql-0.infrastructure-postgresql-ha-postgresql-headless.default.svc.cluster.local port=5432 fallback_application_name=repmgr options=-csearch_path=

--- a/bitnami/postgresql-ha/CHANGELOG.md
+++ b/bitnami/postgresql-ha/CHANGELOG.md
@@ -1,8 +1,13 @@
 # Changelog
 
-## 16.0.5 (2025-05-15)
+## 16.0.6 (2025-05-20)
 
-* [bitnami/postgresql-ha] :zap: :arrow_up: Update dependency references ([#33705](https://github.com/bitnami/charts/pull/33705))
+* [bitnami/postgresql-ha] bugfix: missing password for 'postgres' user when using custom user/db ([#33786](https://github.com/bitnami/charts/pull/33786))
+
+## <small>16.0.5 (2025-05-15)</small>
+
+* [bitnami-postgresql-ha] docs: document security fix on 16.0.0 (#33659) ([9d71b3a](https://github.com/bitnami/charts/commit/9d71b3ad0c5598932670d59c7d118d7a0c496d54)), closes [#33659](https://github.com/bitnami/charts/issues/33659)
+* [bitnami/postgresql-ha] :zap: :arrow_up: Update dependency references (#33705) ([a2fac0d](https://github.com/bitnami/charts/commit/a2fac0de6560264e16c1632a4795a1b111630f0b)), closes [#33705](https://github.com/bitnami/charts/issues/33705)
 
 ## <small>16.0.4 (2025-05-12)</small>
 

--- a/bitnami/postgresql-ha/Chart.yaml
+++ b/bitnami/postgresql-ha/Chart.yaml
@@ -43,4 +43,4 @@ maintainers:
 name: postgresql-ha
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/postgresql-ha
-version: 16.0.5
+version: 16.0.6

--- a/bitnami/postgresql-ha/templates/_helpers.tpl
+++ b/bitnami/postgresql-ha/templates/_helpers.tpl
@@ -87,15 +87,6 @@ Return the PostgreSQL database to create
 {{- end -}}
 
 {{/*
-Return true if PostgreSQL postgres user password has been provided
-*/}}
-{{- define "postgresql-ha.postgresqlPasswordProvided" -}}
-{{- if not (empty (coalesce ((.Values.global).postgresql).postgresPassword .Values.postgresql.postgresPassword) | default "") -}}
-    {{- true -}}
-{{- end -}}
-{{- end -}}
-
-{{/*
 Return the Pgpool Admin username
 */}}
 {{- define "postgresql-ha.pgpoolAdminUsername" -}}

--- a/bitnami/postgresql-ha/templates/postgresql/statefulset.yaml
+++ b/bitnami/postgresql-ha/templates/postgresql/statefulset.yaml
@@ -188,7 +188,7 @@ spec:
               value: {{ .Values.persistence.mountPath | quote }}
             - name: PGDATA
               value: {{ printf "%s/%s" .Values.persistence.mountPath "data" | quote }}
-            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or (include "postgresql-ha.postgresqlCreateSecret" .) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
+            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or (not (include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
             {{- if .Values.postgresql.usePasswordFiles }}
             - name: POSTGRES_POSTGRES_PASSWORD_FILE
               value: "/opt/bitnami/postgresql/secrets/postgres-password"
@@ -488,8 +488,8 @@ spec:
               mountPath: /docker-entrypoint-initdb.d/secret
             {{- end }}
             {{- if .Values.postgresql.usePasswordFiles }}
-            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or (include "postgresql-ha.postgresqlCreateSecret" .) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
-            - name: postgres-password
+            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or (not (include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
+            - name: postgresql-creds
               mountPath: /opt/bitnami/postgresql/secrets/postgres-password
               subPath: postgres-password
             {{- end }}

--- a/bitnami/postgresql-ha/templates/postgresql/statefulset.yaml
+++ b/bitnami/postgresql-ha/templates/postgresql/statefulset.yaml
@@ -188,7 +188,7 @@ spec:
               value: {{ .Values.persistence.mountPath | quote }}
             - name: PGDATA
               value: {{ printf "%s/%s" .Values.persistence.mountPath "data" | quote }}
-            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or (not (include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
+            {{- if not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres") }}
             {{- if .Values.postgresql.usePasswordFiles }}
             - name: POSTGRES_POSTGRES_PASSWORD_FILE
               value: "/opt/bitnami/postgresql/secrets/postgres-password"
@@ -488,7 +488,7 @@ spec:
               mountPath: /docker-entrypoint-initdb.d/secret
             {{- end }}
             {{- if .Values.postgresql.usePasswordFiles }}
-            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or (not (include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
+            {{- if not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres") }}
             - name: postgresql-creds
               mountPath: /opt/bitnami/postgresql/secrets/postgres-password
               subPath: postgres-password

--- a/bitnami/postgresql-ha/templates/postgresql/witness-statefulset.yaml
+++ b/bitnami/postgresql-ha/templates/postgresql/witness-statefulset.yaml
@@ -179,7 +179,7 @@ spec:
               value: {{ .Values.persistence.mountPath | quote }}
             - name: PGDATA
               value: {{ printf "%s/%s" .Values.persistence.mountPath "data" | quote }}
-            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or (not (include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
+            {{- if not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres") }}
             {{- if .Values.witness.usePasswordFiles }}
             - name: POSTGRES_POSTGRES_PASSWORD_FILE
               value: "/opt/bitnami/postgresql/secrets/postgres-password"

